src/jni_core/src/lib.rs

```diff
@@ -896,21 +896,61 @@ extern "system" fn Java_com_risingwave_java_binding_Binding_sendCdcSourceMsgToCh
     })
 }
 
+pub enum SinkWriterStreamRequestEnum {
+    PbRequest(SinkWriterStreamRequest),
+    Chunk {
+        epoch: u64,
+        batch_id: u64,
+        chunk: StreamChunk,
+    },
+}
+
+impl From<SinkWriterStreamRequest> for SinkWriterStreamRequestEnum {
+    fn from(value: SinkWriterStreamRequest) -> Self {
+        Self::PbRequest(value)
+    }
+}
+
 #[no_mangle]
 pub extern "system" fn Java_com_risingwave_java_binding_Binding_recvSinkWriterRequestFromChannel<
     'a,
 >(
     env: EnvParam<'a>,
-    mut channel: Pointer<'a, JniReceiverType<SinkWriterStreamRequest>>,
-) -> JByteArray<'a> {
+    mut channel: Pointer<'a, JniReceiverType<SinkWriterStreamRequestEnum>>,
+) -> JObject<'a> {
     execute_and_catch(env, move |env| match channel.as_mut().blocking_recv() {
         Some(msg) => {
-            let bytes = env
-                .byte_array_from_slice(&Message::encode_to_vec(&msg))
-                .unwrap();
-            Ok(bytes)
+            let obj = match msg {
+                SinkWriterStreamRequestEnum::PbRequest(request) => {
+                    let bytes = env.byte_array_from_slice(&Message::encode_to_vec(&request))?;
+                    call_static_method!(
```

Comment:
It seems there is one additional rust->java call compared to previous implementation. Previously the deserialization step is done in Java side. Not sure the overhead here.

Reply:
I think the overhead here can be ignored because we call it only once per stream chunk. When we iterate over the stream chunk, we will have much more rust->java call, and the overhead of this single call should not be considerable compared to those calls.